docs/pages/framework/testing-cheat-codes.md

```diff
@@ -0,0 +1,242 @@
+
+
+1. `set_block_number`
+
+This cheat code helps you set the current block number to the specified value, allowing you to simulate different block heights for testing purposes.
+
+This one is helpful to test a contract's behavior at a specific block height, such as checking if a certain function is only callable after a certain block number.
+
+```
+use starknet::{testing, get_block_number};
+
+testing::set_block_number(1234567);
+
+assert!(get_block_number() == 1234567, 'bad block number');
+```
+
+2. `set_caller_address`
+
+This cheat code helps you set the caller address to the provided contract address, enabling you to test contract interactions with different callers. This cheat code can be applied when:
+
+- Testing a contract's access control mechanisms, such as only allowing certain addresses to call specific functions.
+- Simulating a scenario where a contract is called by a different address.
+
+```
+use starknet::{testing, get_caller_address, contract_address_const};
+
+let user_one = contract_address_const::<'user1'>();
+testing::set_caller_address(user_one);
+assert(get_caller_address() == user_one, 'bad caller';
+```
+
+3. `set_contract_address`
+
+This cheat code helps you set the contract address to the provided value, allowing you to test contract deployment and interactions.
+
+It is important to note that any test function is considered a contract, which by default uses the `0` address. Using `set_contract_address` allows you to mock the current address of the testing function, making it useful to call other contract that may use `get_caller_address`.
+
+```
+use starknet::{testing, get_contract_address, contract_address_const};
+
+const HUB_ADDRESS: ContractAddress = contract_address_const::<'hub'>()
+
+testing::set_contract_address(HUB_ADDRESS);
+
+assert(get_contract_address() == HUB_ADDRESS, 'BAD ADDRESS');
+```
+
+4. `set_block_timestamp`
+
+This cheat code helps you set the block timestamp to the specified value, allowing you to test contract behavior at different points in time. You can apply this code when:
+
+- Testing a contract's behavior at a specific point in time, such as checking if a certain function is only callable during a certain time period.
+- Simulating a scenario where a contract is deployed at a different point in time.
+
+```
+use starknet::{testing, get_block_timestamp};
+
+testing::set_block_timestamp(123456);
+
+assert(get_block_timestamp == 123456, 'bad timestamp');
+
+```
+
+5. `set_version`
+
+This cheat code helps one set the version to the provided value, enabling you to test contract behavior with different versions. You can apply this when:
+
+- Testing a contract's behavior with different versions, such as checking if a certain function is only callable in a specific version.
+- Simulating a scenario where a contract is upgraded to a different version.
+
+```
+use starknet::{testing, get_version};
```

Comment:
```suggestion
use starknet::{testing, get_tx_info};
```
`get_version` function doesn't exist. It's the version of the transaction inside the transaction execution info:
```rust
testing::set_version(1);
assert_eq!(get_tx_info().unbox().version, 1_felt252);
```